Userland/Applications/FileManager/DirectoryView.cpp

```diff
@@ -54,6 +54,12 @@ static HashTable<RefPtr<GUI::Window>> file_operation_windows;
 
 static void run_file_operation([[maybe_unused]] FileOperation operation, const String& source, const String& destination, GUI::Window* parent_window)
 {
+    struct stat st = {};
+    if (stat(source.characters(), &st) < 0) {
+        perror("stat");
+        return;
+    }
```

Comment:
Let's put this responsibility in the `FileOperation` helper program and have it signal an error to `FileManager` instead.

Reply:
That seems like the better approach.